sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIOTest.java

```diff
@@ -1935,3 +1965,21 @@ abstract static class Builder {
     }
   }
 }
+
+class FakeResourceStatsSupplier implements BigtableIO.ResourceStatsSupplier, Serializable {
```

Comment:
please keep these as private static classes inside the test case class

Reply:
Also to make it easier to track down where magic numbers are coming from I would add a ctor that takes the fake cumulativeThrottlingTimeNanos and passes it down into FakeResourceStats. This way at the call site you can see where the numbers are coming from